pkg/apis/application/v1alpha1/types.go

```diff
@@ -1011,44 +1015,74 @@ func (repo *Repository) IsLFSEnabled() bool {
 	return repo.EnableLFS
 }
 
-func (m *Repository) CopyCredentialsFrom(source *Repository) {
+func (repo *Repository) CopyCredentialsFrom(source *Repository) {
 	if source != nil {
-		if m.Username == "" {
-			m.Username = source.Username
+		if repo.Username == "" {
+			repo.Username = source.Username
 		}
-		if m.Password == "" {
-			m.Password = source.Password
+		if repo.Password == "" {
+			repo.Password = source.Password
 		}
-		if m.SSHPrivateKey == "" {
-			m.SSHPrivateKey = source.SSHPrivateKey
+		if repo.SSHPrivateKey == "" {
+			repo.SSHPrivateKey = source.SSHPrivateKey
 		}
-		m.InsecureIgnoreHostKey = m.InsecureIgnoreHostKey || source.InsecureIgnoreHostKey
-		m.Insecure = m.Insecure || source.Insecure
-		m.EnableLFS = m.EnableLFS || source.EnableLFS
-		if m.TLSClientCertData == "" {
-			m.TLSClientCertData = source.TLSClientCertData
+		repo.InsecureIgnoreHostKey = repo.InsecureIgnoreHostKey || source.InsecureIgnoreHostKey
+		repo.Insecure = repo.Insecure || source.Insecure
+		repo.EnableLFS = repo.EnableLFS || source.EnableLFS
+		if repo.TLSClientCertData == "" {
+			repo.TLSClientCertData = source.TLSClientCertData
 		}
-		if m.TLSClientCertKey == "" {
-			m.TLSClientCertKey = source.TLSClientCertKey
+		if repo.TLSClientCertKey == "" {
+			repo.TLSClientCertKey = source.TLSClientCertKey
 		}
-		if m.TLSClientCAData == "" {
-			m.TLSClientCAData = source.TLSClientCAData
+		if repo.TLSClientCAData == "" {
+			repo.TLSClientCAData = source.TLSClientCAData
 		}
 	}
 }
 
-type Repositories []*Repository
+func (repo *Repository) GetGitCreds() git.Creds {
+	if repo == nil {
+		return git.NopCreds{}
+	}
+	if repo.Username != "" && repo.Password != "" {
```

Comment:
I think this should also test for TLS client certificates, so the correct check should be:
```
if (repo.Username != "" && repo.Password != "") || repo.TLSClientCertData != "" {
}
```

Reply:
fixed

Reply:
Had to revert this change temporarily. Because of this change  in some cases we incorrectly use https creds instead of ssh. Working on fix in the separate PR

Reply:
The types.go package already depends on multiple utility packages: grpc/ kubernetes, kubernetes cli. I agree it might be an issue in some cases but this is the way it is developed from day 1.